src/TrueLayer/MerchantAccounts/MerchantAccountsApi.cs

```diff
@@ -83,8 +83,50 @@ public async Task<ApiResponse<GetPaymentSourcesResponse>> GetPaymentSources(stri
             }
 
             return await _apiClient.GetAsync<GetPaymentSourcesResponse>(
-                _baseUri.Append($"merchant-accounts/{merchantAccountId}/payment-sources?user_id={userId}"),
+                _baseUri.Append($"{merchantAccountId}/payment-sources?user_id={userId}"),
+                authResponse.Data!.AccessToken,
+                cancellationToken: cancellationToken
+            );
+        }
+
+        /// <inheritdoc />
+        public async Task<ApiResponse<GetTransactionsResponse>> GetTransactions(
+            string merchantAccountId,
+            DateTimeOffset from,
+            DateTimeOffset to,
+            bool usePagination = true,
+            string? cursor = null,
+            string? type = null,
+            CancellationToken cancellationToken = default)
+        {
+            merchantAccountId.NotNullOrWhiteSpace(nameof(merchantAccountId));
+            merchantAccountId.NotAUrl(nameof(merchantAccountId));
+            to.GreaterThan(from, nameof(to), nameof(from));
+            cursor.NotAUrl(nameof(cursor));
```

Comment:
Why are we validating this?

Reply:
For security reasons. A few months ago we had a security breach where consumers of the SDK could inject URLs into string parameters. This check prevents it.

Reply:
Ah, good to know! Is this validated also at the API level if you know the answer? I think we could replicate this on the Java lib as well, I just would like to know how urgent it is (This does not help with the case where a malicious user use a previous lib version...)

Reply:
TBH I am not aware of any issues at api level. Maybe it is worth implementing it in the Java lib as well,